docs/world-compat-details/index.md

```diff
@@ -365,4 +365,334 @@ LBT 扩展指令集，`magic` = `0x42540001`
+       been … received as the third argument to a signal handler”
+[^4]: 对比 [`sigaction`](https://elixir.bootlin.com/glibc/glibc-2.38/source/sysdeps/unix/sysv/linux/bits/sigaction.h#L27) 和
+       [`kernel_sigaction`](https://elixir.bootlin.com/glibc/glibc-2.38/source/sysdeps/unix/sysv/linux/kernel_sigaction.h#L9)
+
+截至 glibc 2.38 版本，新世界的 `*context` 函数仅能处理 `ucontext_t` 结构体中的通用（整数）寄存器部分，而不能处理浮点、向量扩展和 LBT 扩展部分。
+旧世界的 `*context` 函数能处理 `ucontext_t` 结构体中通用（整数）寄存器和浮点寄存器，但不能处理向量扩展和 LBT 扩展部分。
+此外，旧世界 glibc 提供的 `ucontext_t` 定义[^5]与旧世界内核[^6]的定义对比，可以发现前者缺少用于存放 LBT 扩展寄存器的 `uc_mcontext.sc_scr[4]` 字段，
+致使后续用于存放浮点寄存器的 `uc_mcontext.sc_fpregs[32]` 字段的偏移量发生了变化。这意味着，
+旧世界的 `*context` 函数在处理旧世界内核提供给信号处理函数的 `ucontext_t` 结构体时，无法正确处理浮点寄存器。
+综上所述，旧世界 glibc 也仅能正确处理 `ucontext_t` 结构体的通用（整数）寄存器部分。这样，在功能上，
+恰好新旧世界达成了一致——都只能正确处理通用寄存器部分。
+
+[^5]: 可以在 Loongnix 发行版的 `/usr/include/loongarch64-linux-gnu/sys/ucontext.h` 找到
+[^6]: 可以在 Loongnix 发行版的 `/usr/include/loongarch64-linux-gnu/bits/sigcontext.h` 找到
+
+glibc 提供了两个函数，可以被用于注册信号处理函数，这两个函数分别是 `sigaction` 和 `signal`。
+其中，`signal` 函数是对 `sigaction` 函数的封装。这样，无论是如何注册的信号处理函数，
+都会在第三个参数中接受到一个 `ucontext_t` 结构体指针。但是根据 glibc [文档](https://man7.org/linux/man-pages/man2/signal.2.html)，
+`signal` 函数注册的信号处理函数只应该接受第一个参数。这意味着，如果 `signal` 函数注册的信号处理函数
+遵循了文档的要求，只接受第一个参数，那么它会忽略第三个参数，即`ucontext_t` 结构体指针会被忽略。
+这意味着，`signal` 函数注册的信号处理函数，是新旧世界无关的，不需要针对这样的信号处理函数做任何兼容性处理。
+而 `sigaction` 函数注册的信号处理函数，是新旧世界相关的，需要额外的兼容性处理。一种可能的兼容性处理是，
+当用户调用 `sigaction` 函数注册一个接受旧世界的 `ucontext_t` 结构体信号处理函数时，实际不注册这个函数，
+而是注册一个包裹函数，这个包裹函数接受新世界的 `ucontext_t` 结构体，并在栈上相应构造一个旧世界的 `ucontext_t` 结构体，
+然后调用用户提供的信号处理函数；在用户提供的信号处理函数返回后，将旧世界的 `ucontext_t` 结构体的内容拷贝回新世界的 `ucontext_t` 结构体中。
+需要注意的是，信号处理函数无法接受自定义的参数，这意味着原始用户提供的信号处理函数的地址必须以某种方式保存下来，
+以便包裹函数调用它。而在实现这一点时要十分小心地处理锁和信号屏蔽，因为整个这个注册信号处理函数的过程不再是原子的。
+
+无论如何处理，都不会改变新旧世界 `ucontext_t` 结构体完全不兼容的事实，也无法扭转旧世界 glibc
+和旧世界内核 `ucontext_t` 结构体存在差异的现状，因此如果出现了新旧世界可执行程序和动态连接库的混合连接，
+并且存在将 `ucontext_t` 结构体在新旧世界间传递的情况，始终无法保证正确识别 `ucontext_t` 结构体是新世界的还是旧世界的。
+不过幸运的事，这样的情况十分罕见。
+
+下列函数涉及 `ucontext_t` 结构体，因此新旧世界的函数完全不二进制兼容：
+
+- `getcontext`
+- `setcontext`
+- `makecontext`
+- `swapcontext`
+
+下列函数涉及注册能接收 `ucontext_t` 结构体的信号处理函数，因此需要额外的兼容性处理：
+
+- `sigaction`
+
+下列函数注册的信号处理函数会忽略 `ucontext_t` 结构体指针，因此无需特殊处理：
+
+- `signal`
+
+#### longjmp 相关
+
+`setjmp`、`sigsetjmp`、`longjmp`、`siglongjmp` 函数是用于实现非局部跳转的函数。
+其中 `set*jmp` 函数会将当前的程序状态（可选地，包括当前的信号掩码，即 sigmask）保存到一个 `jmp_buf` 结构体中，
+而 `*longjmp` 函数则将上述状态恢复。这些函数的行为与 `*context` 函数类似。在新旧世界中，
+`jmp_buf` 结构体的定义完全一致，因此这些 `*jmp` 函数是完全二进制兼容的。
+
+下列函数涉及 `jmp_buf` 结构体，在新旧世界间完全兼容：
+
+- `setjmp`
+- `sigsetjmp`
+- `longjmp`
+- `siglongjmp`
+
+#### `fstat` 相关
+
+新世界的内核中，缺少 `fstat` 和 `newfstatat` 系统调用。这两个系统调用可以被用于获取文件的元数据。
+其中，`fstat` 可以获取一个打开的文件描述符 （file descriptor, fd）所对应的文件的元数据；
+而 `newfstatat` 则既可以获取一个打开的文件描述符所对应的文件的元数据，也可以获取一个路径所对应的文件的元数据。
+取代了这两个系统调用的是 `statx`。`statx` 的功能与 `newfstatat` 的功能基本一致，
+但是可以按需提供更多的信息。
+
+在 2.38 的 glibc 中，所有 `*stat*` 函数会在编译期检查内核是否提供了 `fstat` 或 `newfstatat`，
```

Comment:
Reworded a little to prevent a potential misunderstanding: no probing of live kernel is done. "编译" can both mean the actual compilation i.e. `make` process, or the whole `configure && make` process in casual Chinese speech, so this can confuse readers not paying attention.

```suggestion
在 2.38 的 glibc 中，所有 `*stat*` 函数会在编译期检查内核的头文件是否声明了 `fstat` 或 `newfstatat` 存在，
```

Reply:
其实是通过 `#ifdef` 检查的

Reply:
> 其实是通过 `#ifdef` 检查的

确实，我前边提议的改法其实也没明确具体怎么个「声明」法；感觉要不要改见仁见智！